src/main/java/redis/clients/jedis/BinaryJedisCluster.java

```diff
@@ -14,14 +14,16 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.regex.Pattern;
 
 public class BinaryJedisCluster implements BasicCommands, BinaryJedisClusterCommands,
-    MultiKeyBinaryJedisClusterCommands, JedisClusterBinaryScriptingCommands, Closeable {
+    MultiKeyBinaryJedisClusterCommands , HighLevelBinaryJedisClusterCommands, JedisClusterBinaryScriptingCommands, Closeable {
 
   public static final short HASHSLOTS = 16384;
   protected static final int DEFAULT_TIMEOUT = 2000;
   protected static final int DEFAULT_MAX_REDIRECTIONS = 5;
-
+  protected static final Pattern SCAN_VALID_MATCH_REGEX_PATTERN = Pattern.compile("\\{[^\\{^\\}]+\\}\\*");
```

Comment:
@marcosnils  I only use the regex to validate input since only {TAG}* should be supported in cluster mode . I can disable the limitation though it might create confusion amongst the user as to why this is not working in all scenarios

I will add cluster based scanning implementation that does distributed scanning ( supporting all cases ) in the future but for now only a speciic case is covered